src/main/java/org/openjdk/jextract/impl/ClassSourceBuilder.java

```diff
@@ -235,42 +176,11 @@ void emitDocComment(Type.Function funcType, String name) {
         append(" */\n");
     }
 
-    // is the name enclosed enclosed by a class of the same name?
-    boolean isEnclosedBySameName(String name) {
-        return className().equals(name) ||
-                (isNested() && enclosing.isEnclosedBySameName(name));
-    }
-
-    protected void emitPackagePrefix() {
-        if (!isNested()) {
-            assert packageName().indexOf('/') == -1 : "package name invalid: " + packageName();
-            append("// Generated by jextract");
-            if (SHOW_GENERATING_CLASS) {
-                append(" (via ");
-                append(getClass().getName());
-                append(")");
-            }
-            append("\n\n");
-            if (!packageName().isEmpty()) {
-                append("package ");
-                append(packageName());
-                append(";\n\n");
-            }
-        }
-    }
-
-    protected void emitImportSection() {
-        if (!isNested()) {
-            append("import java.lang.invoke.MethodHandle;\n");
-            append("import java.lang.invoke.VarHandle;\n");
-            append("import java.nio.ByteOrder;\n");
-            append("import java.lang.foreign.*;\n");
-            append("import static java.lang.foreign.ValueLayout.*;\n");
-        }
-    }
-
-    void emitConstantGetter(String mods, String getterName, boolean nullCheck, String symbolName, Constant constant) {
+    final void emitConstantGetter(String mods, String getterName, boolean nullCheck, String symbolName, Constant constant, Declaration decl) {
```

Comment:
This method is only used by `Constants,Builder` so maybe it should be moved there. I've left it for now.